scripts/update-delay/recipe_times.py

```diff
@@ -0,0 +1,316 @@
+        for b in st.blobs:
+            if b.name == 'meta.yaml':
+                # both the hex hash and the actual data are kept in the cache.
+                if b.hexsha not in cache:
+                    bout = BytesIO()
+                    b.stream_data(bout)
+                    data = bout.getvalue()
+                    if data not in cache:
+                        m = extract_meta(data)
+                        cache[data] = m
+                    else:
+                        m = cache[data]
+                    cache[b.hexsha] = m
+                reptime = cache[b.hexsha]
+                if reptime is not None:
+                    res.append(reptime)
+    return res
+
+
+def walk_repo(repo, init_commit=None):
+    '''Iterate over all commits upstream of `init_commit``'''
+    if init_commit is None:
+        queue = [repo.head.commit]
+    else:
+        queue = [repo.commit(init_commit)]
+    seen = set()
+    while queue:
+        c = queue.pop()
+        if c not in seen:
+            seen.add(c)
+            queue.extend(c.parents)
+            yield c
+
+@TaskGenerator
+def retrieve_earliest_commit():
+    repo = git.Repo(environ["BIOCONDA_RECIPES_DIRECTORY"])
+    cache = {}
+    earliest = {}
+    initial = True
+    for c in walk_repo(repo, '6ef812f1b808ac7dc39313ff133ed42f027bebdb'):
+        timestamp = c.committed_date
+        for repinfo in extract_recipes(c, cache):
+            if repinfo.pname in earliest:
+                cur = earliest[repinfo.pname]
+                if cur[0].version == repinfo.version:
+                    if cur[1] > timestamp:
+                        cur[1] = timestamp
+                elif version_greater(repinfo.version, cur[0].version):
+                    if initial:
+                        earliest[repinfo.pname] = [repinfo, timestamp]
+            else:
+                earliest[repinfo.pname] = [repinfo, timestamp]
+        initial = False
+    data = [(p.pname, p.version, p.url, p.source, t) for p, t in earliest.values()]
+    return data
+
+@TaskGenerator
+def get_pypi_metadata(package, version):
+    import requests
+    import json
+    import time
+    time.sleep(.1) # Do not bombard the server
+    return json.loads(requests.get('https://pypi.python.org/pypi/{}/{}/json'.format(package, version)).text)
+
+@TaskGenerator
+def get_github_releases_metadata(owner, repo, fetch_tags=False):
+    import requests
+    import json
+    import time
+    time.sleep(.1) # Do not bombard the server
+    if 'GITHUB_USERNAME' not in environ or \
+        'GITHUB_TOKEN' not in environ:
+        from sys import exit
+        print("Please set the environmental variables GITHUB_USERNAME and GITHUB_TOKEN to query Github's API")
+        exit(1)
+
+    what = ('release' if not fetch_tags else 'tags')
+    r = requests.get('https://api.github.com/repos/{}/{}/{}'.format(owner, repo, what),
+                auth=(environ['GITHUB_USERNAME'], environ['GITHUB_TOKEN']))
```

Comment:
You can use pygithub to access the github API. Please see my collect-pr-data script: https://github.com/bioconda/bioconda-paper/blob/master/scripts/collect-pr-data.py